tools/boost_names.py

```diff
@@ -117,28 +137,36 @@ def get_libraries(jamfile: Path) -> T.List[BoostLibrary]:
             if not lname.startswith('boost_'):
                 continue
 
+            # Count `:` to only select the 'usage-requirements'
+            # See https://boostorg.github.io/build/manual/master/index.html#bbv2.main-target-rule-syntax
+            colon_counter = 0
+            usage_requirements: T.List[str] = []
+            for j in parts:
+                if j == ':':
+                    colon_counter += 1
+                elif colon_counter >= 4:
+                    usage_requirements += [j]
+
             # Get shared / static defines
             shared: T.List[str] = []
             static: T.List[str] = []
             single: T.List[str] = []
             multi: T.List[str] = []
-            for j in parts:
+            for j in usage_requirements + project_usage_requirements:
```

Comment:
You might consider `itertools.chain` instead of concatenating lists, to avoid building a concrete list.

Reply:
Thanks, I will keep it in mind for the next time, but in this case, I would go for the better readability with the `+`, since I don't expect any of the lists to exceed 10 elements and this is a code generator which is only run once every boost release (at most).